### PR TITLE
feat: Enable Geo Blocking for restricted territories

### PR DESCRIPTION
**Pull request by cpcramer**
**What changed? Why?**
Add Vercel Edge Middleware to enable Geo Blocking for the entire OnchainKit site. 
- Add restricted page. 
- Redirect users in a blocked region to the restricted page.
- Restrict the following countries `North Korea (KP), Iran (IR), Syria (SY), Cuba (CU), Crimea (UA-43), Luhansk (UA-09), Donetsk (UA-14)`

Edge Middleware is code that executes before a request is processed on a site. We can further update our middleware to execute custom logic, rewrite, redirect, add headers and more, before returning a response!

[API Reference.](https://vercel.com/docs/functions/edge-middleware/middleware-api#middleware-helper-methods)

Screenshot 2024-06-18 at 1 08 02 PM

**Notes to reviewers**

**How has it been tested?**
Tested in dev with the US being blocked and unblocked.

### CODE REVIEW

File: site/middleware.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+  if (blockedCountries.includes(country)) {
+    return new NextResponse('Access Denied', { status: 403 });
+  }
+
+  return NextResponse.next();
+}
+
+export const config = {
+  matcher: '/swap/swap*', // Apply middleware to all paths
```
Comment (Zizzamia) [inline site/middleware.ts:17]: Let's test this PR first with a testing page.

And then we use it for the entire site.

File: site/docs/pages/blocked.tsx [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export default function Login() {
+  return (
+    <div>
+      <h1>You have been Blocked</h1>
```
Comment (Zizzamia) [inline site/docs/pages/blocked.tsx:4]: Do we have a more correct message? `Blocked` seems a bit affermative, can you check with other teams what kind of messaging they use.

Comment (cpcramer) [reply site/docs/pages/blocked.tsx:4]: Yes! This is 100% just for locally testing to make sure the redirect is working properly for me!

File: site/middleware.ts (hunk 1 above)
Comment (cpcramer) [reply site/middleware.ts:17]: Sounds good!

File: site/middleware.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  'CU',
+  'UA-43',
+  'UA-09',
+  'UA-14',
+  'UNKNOWN',
+];
+
+export const config = {
+  matcher: ['/'],
+};
```
Comment (cpcramer) [inline site/middleware.ts:18]: Currently set to run on the entire site

File: site/middleware.ts (hunk 1 above)
Comment (cpcramer) [reply site/middleware.ts]: It's been tested in Vercel Staging. Would you prefer to test a single page (/swap/swap) in prod as well before enabling for the entire website?

File: site/middleware.ts (hunk 3 above)
Comment (cpcramer) [reply site/middleware.ts:18]: If we'd like to test a single endpoint in prod before rolling out to the entire site then I can update this matcher to be `/swap/swap`

Comment (Zizzamia) [reply site/middleware.ts:18]: Perfect, can you set the matcher to a demo-page. So we can merge this.

Comment (cpcramer) [reply site/middleware.ts]: Done! Set to a demo page at /temp-demo

File: site/docs/pages/temp-demo.mdx [hunk 4]
```diff
@@ -0,0 +1,7 @@
+
+<div className="flex justify-center p-4">
+  <div className="text-center">
+    <h1 className="text-3xl md:text-4xl">Temp Demo Page</h1>
+    <p className="text-lg md:text-xl">Temporary demo page to test geo blocking restrictions.</p>
+  </div>
+</div>
```
Comment (cpcramer) [inline site/docs/pages/temp-demo.mdx:7]: Temporary - to be deleted once we've verified the geo blocking feature in production.

**Review by ilikesymmetry — APPROVED** [review]

**Pull request merged**